### ISSUE TITLE
[Issue]: Jellyfin looks to the wrong path

### ISSUE THREAD

**Issue by Germs2004** [jellyfin/jellyfin#7576]
### Please describe your bug

During installation on Windows 11, I chose to install as a Windows Service and run as Network Service.  I set my data path to D:\Jellyfin\, and most things work fine.  But Jellyfin still looks at C:\ProgramData\Jellyfin for some things, which causes errors and causes the "Scan Media Library" button to always fail immediately.  I can't find anything in my config that points to ProgramData, so I think this may be a bug in the code where it is hard coded to use ProgramData.

In the log, attached, you see Jellyfin tries to read from `C:\ProgramData\Jellyfin\Server\root\default\Movies` but the correct path should be `D:\Jellyfin\root\default\Movies`

I tried to track down the bug and got as far as MediaBrowser.Controller.Entities.CollectionFolder.CreateResolveArgs()
In that function, the first line says
`var path = ContainingFolderPath;`
That "ContainingFolderPath" variable points to C:\ProgramData.  If someone can trace that back to figure out why it's finding C:\ProgramData instead of my D:\Jellyfin, I bet that will solve the bug.

A similar, cosmetic bug is that Jellyfin creates C:\ProgramData\Jellyfin\Server\transcodes\ every time it starts up instead of using D:\Jellyfin\transcodes\.

I saw another user had the problem but they never found a proper fix: https://github.com/jellyfin/jellyfin/issues/7464

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: Windows 11
- Virtualization: no
- Browser: Chrome
```

### Jellyfin logs

```shell
[2022-04-09 22:32:03.452 -04:00] [ERR] [15] MediaBrowser.Controller.Entities.BaseItem: Error refreshing owned items for "C:\ProgramData\Jellyfin\Server\root\default\Movies"
System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\ProgramData\Jellyfin\Server\root\default\Movies'.
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, Boolean isNormalized, EnumerationOptions options)
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.FileSystemInfos(String directory, String expression, EnumerationOptions options, Boolean isNormalized)
   at System.IO.DirectoryInfo.InternalEnumerateInfos(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.DirectoryInfo.EnumerateFileSystemInfos(String searchPattern, EnumerationOptions enumerationOptions)
   at Emby.Server.Implementations.IO.ManagedFileSystem.GetFileSystemEntries(String path, Boolean recursive)
   at MediaBrowser.Controller.Providers.DirectoryService.<>c.<GetFileSystemEntries>b__5_0(String p, IFileSystem fileSystem)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd[TArg](TKey key, Func`3 valueFactory, TArg factoryArgument)
   at MediaBrowser.Controller.Providers.DirectoryService.GetFileSystemEntries(String path)
   at MediaBrowser.Controller.IO.FileData.GetFilteredFileSystemEntries(IDirectoryService directoryService, String path, IFileSystem fileSystem, IServerApplicationHost appHost, ILogger logger, ItemResolveArgs args, Int32 flattenFolderDepth, Boolean resolveShortcuts)
   at MediaBrowser.Controller.Entities.CollectionFolder.CreateResolveArgs(IDirectoryService directoryService, Boolean setPhysicalLocations)
   at MediaBrowser.Controller.Entities.CollectionFolder.GetFileSystemChildren(IDirectoryService directoryService)
   at MediaBrowser.Controller.Entities.BaseItem.RefreshMetadata(MetadataRefreshOptions options, CancellationToken cancellationToken)
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by Germs2004** [comment]
I have found the answer.  Open data\library.db with a sqlite editor, go to table "TypedBaseItems", and find all the rows with these "type" values:
```
MediaBrowser.Controller.Entities.UserRootFolder
MediaBrowser.Controller.Entities.CollectionFolder
```
In my case, those rows had the wrong "path" values in them.  The fix is easy: stop the Jellyfin service, edit those paths in the db, and start the service.

**Comment by myfingerhurt** [comment]
Thank your @Germs2004.

I had the same issue, I also was trying to find path in `library.db`. 
But every time I was trying to open table `TypedBaseItems`, 
my SQLite crashes, so I hadn't been able to find those remaining paths.

Here is what I do this time.
My jellyfin data path was at
`C:\Users\nas\AppData\Local\jellyfin\`
Now I have moved them to
`C:\_Tools\jellyfin\data`
Also I would like to run Jellyfin with less privileged user and limited access to my files.
So I created a shortcut and put it into `shell:statup`
`C:\Windows\System32\runas.exe /savecred /user:emby "C:\_Tools\jellyfin\jellyfin.exe --datadir C:\_Tools\jellyfin\data"`
After replacing old path with new one in `.xml` files under this path
`C:\_Tools\jellyfin\data\config`

Had to replace all path remains in  `library.db`, with help of  `sqlite_web`.
`sqlite_web library.db`
Make sure there are existing
```
SELECT *
FROM "TypedBaseItems" 
WHERE path like "C:\Users\nas%"
```
Changing the path
```
UPDATE "TypedBaseItems" 
SET Path=REPLACE(Path, "C:\Users\nas\AppData\Local\jellyfin", "C:\_Tools\jellyfin\data")
WHERE path like "C:\Users\nas\AppData\Local\jellyfin%"
```
Verifying the result, should be found nothing.
```
SELECT *
FROM "TypedBaseItems" 
WHERE path like "C:\Users\nas%"
```

**Comment by DQJeremy** [comment]
I had the same error for example: MediaBrowser.Controller.Entities.BaseItem: Error refreshing owned items for "C:\ProgramData\Jellyfin\server\data\root\default\Photos"
I had moved everything to the D: drive and had nothing pointed to the C: drive.
What I did to resolve it was to create a symbolic link from D:\ProgramData\Jellyfin to C:\ProgramData\Jellyfin.
This seemed easier to me to do than to edit the values in the database.

After I did this, Scan Libraries started working again.
